core/src/main/scala/kafka/tools/DumpLogSegments.scala

```diff
@@ -85,7 +85,7 @@ object DumpLogSegments {
 
     val files = options.valueOf(filesOpt).split(",")
     val maxMessageSize = options.valueOf(maxMessageSizeOpt).intValue()
-    val isDeepIteration = options.has(deepIterationOpt)
+    val isDeepIteration = options.has(deepIterationOpt) || printDataLog
```

Comment:
Do you think we should also imply deep iteration if a message decoder is provided explicitly?

Reply:
@hachikuji `printDataLog` is already automatically set if decoders are provided. So this will enable deep iteration in that case.

Reply:
Ah, you're right. I was looking for it in the logic below when constructing the message parser.